### ISSUE TITLE
Non-linear release histories not supported

### ISSUE THREAD

**Issue by codysoyland** [skywinder/github-changelog-generator#497]
I may have missed a flag or something, but this doesn't appear to support release histories that include multiple maintained release branches.

As an example, see the changelog generated for github.com/pilosa/pilosa: https://gist.github.com/codysoyland/703101d5be09fc811cb3f7a6ab8d1e8e

As you can see, we made releases in the following order:

1. 0.2.4
2. 0.3.1
3. 0.2.5
4. 0.3.2

However, the changelog assumes that each of these releases was an evolution from the previous by order of date. But we are actually maintaining two release branches, a 0.2.x branch and a 0.3.x branch.

In other words, the previous version that is used to generate the change list and diff link should not be the previous tag by date, but the most recent tagged ancestor in git history.

**Comment by Slamdunk** [comment]
Totally 👍 for this.

Maybe also related to https://github.com/skywinder/github-changelog-generator/issues/256

Supporting two or more branches is common, bugfixes go to the lowest supported branch and then merged in more recent branches, while latest branch keeps getting PR merged in the meantime.

This bug mixes the result that therefore must be manually cleared 😞

**Comment by hunner** [comment]
#619 fixes this for PR merge detection but doesn't fix it for the generated diff links. Something like https://github.com/skywinder/github-changelog-generator/pull/619#issuecomment-370598725 would be needed.

**Comment by mvz** [comment]
It is unclear to me what the desired result is. Should new changes in the 0.2.x series even appear in the 0.3.x changelog? If not, the `--release-branch` option introduced in #305 should help.
